firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanel.swift

```diff
@@ -350,6 +350,10 @@ class HistoryPanel: UIViewController,
             }
 
             showClearRecentHistory()
+        case .OpenRecentlyClosedTabs:
+            // TODO: Need to also have the history panel show up if not active
+            historyCoordinatorDelegate?.showRecentlyClosedTab()
```

Comment:
So while this does work, it doesn't open up the panel...I tried timeboxing figuring out an easier way to deal with this but might need some help from the iOS team on this!

Reply:
@skhamis Just to make sure I'm understanding: is the history panel not opening at all? What happens if you manually open it (Settings > History), then close it, then attempt to run the test? 

The `HistoryPanel` isn't actually initialized until the first time the UI is opened (which typically happens in `LibraryCoordinator.makeChildPanels()`, so it's possible that there isn't an active instance of it which can respond to this notification.

If the above steps (manually opening the history UI, then closing, then testing the notification) fix the problem then it's very likely due to it simply not having been initialized yet.

Reply:
The history panel is not opening at all. Even opening it -> closing won't open it up. However when I manually go to Settings -> History _after_ tapping the notification, it's already on the recently closed tab page. So I think the routing is working but just not opening the panel!

Reply:
@skhamis So I could be misunderstanding, but I suspect that the issue might be that we need to push this UI from a point higher up in the coordinator hierarchy, possibly via something like `BrowserCoordinator.showLibrary(with: .history)` and then after that we can ask the history panel itself to push to the Recently Closed Tabs.

Reply:
> possibly via something like BrowserCoordinator.showLibrary(with: .history) and then after that we can ask the history panel itself to push to the Recently Closed Tabs.

Thanks @mattreaganmozilla I think I was looking for something like that! I think it's in a good place now with the panel showing up and recently closed tabs also showing when the notification is tapped!